Contain oboarding information in a scroll view

johnny9 opened the pull request:
Addresses https://github.com/bitcoin-core/gui-qml/issues/186


[![Windows](https://img.shields.io/badge/OS-Windows-green)](https://api.cirrus-ci.com/v1/artifact/github/bitcoin-core/gui-qml/win64/insecure_win_gui.zip?branch=pull/194)
[![Intel macOS](https://img.shields.io/badge/OS-Intel%20macOS-green)](https://api.cirrus-ci.com/v1/artifact/github/bitcoin-core/gui-qml/macos/insecure_mac_gui.zip?branch=pull/194)
[![Apple Silicon macOS](https://img.shields.io/badge/OS-Apple%20Silicon%20macOS-green)](https://api.cirrus-ci.com/v1/artifact/github/bitcoin-core/gui-qml/macos_arm64/insecure_mac_arm64_gui.zip?branch=pull/194)
[![ARM64 Android](https://img.shields.io/badge/OS-Android-green)](https://api.cirrus-ci.com/v1/artifact/github/bitcoin-core/gui-qml/android/insecure_android_apk.zip?branch=pull/194)

hebasto commented on src/qml/controls/OnboardingInfo.qml:
```diff
@@ -30,55 +30,64 @@ Item {
+        contentWidth: width
+        contentHeight: information.height + continueButton.height
+            + continueButton.anchors.topMargin + continueButton.anchors.bottomMargin
+        
```
Extra spaces

```suggestion

```

must be removed here to make `test/lint/lint-whitespace.py` happy :)

hebasto reviewed (commented):
Suggesting to rebase to fix CI tasks.

jarolrod reviewed (commented):
This is the better option at least when considering how it reacts to the gui resized within a tiling window manager (which doesn't respect the minimum sizes like compositing window managers)

In the below screenshows i tried to scroll all the way down to the bottom to be able to access the continue button, could only do so with this option (option2):


| option1 | option2 |
| ------- | ------- |
| ![option-1-screen](https://user-images.githubusercontent.com/23396902/206110118-daad0d2e-439e-4831-9207-c10190e5a6e8.png) | ![option2-screen](https://user-images.githubusercontent.com/23396902/206110132-a4fc0563-9139-4ecb-baf2-37aeec084bff.png) |

Have not yet tested on android.

johnny9 commented:
Rebase from[ 19df4a5 ](https://github.com/bitcoin-core/gui-qml/commit/19df4a52e876aee997ef90a3b3a0680678b1c065)to[ fd23e05](https://github.com/bitcoin-core/gui-qml/commit/fd23e05dd5f4d7f4db98ab127300eb9ae9c8d55b):
  - Fixed whitespace/lint error

hebasto commented:
Rebase?

johnny9 commented:
Updated branch from[ fd23e05 ](https://github.com/bitcoin-core/gui-qml/commit/fd23e05dd5f4d7f4db98ab127300eb9ae9c8d55b)to[ b8cf76](https://github.com/bitcoin-core/gui-qml/commit/b8cf76c7c2279966c898a244d08de3757bd0eb4e)

  - rebased with main

hebasto commented:
Needs rebase again.

johnny9 commented:
Update from[ b8cf76c ](https://github.com/bitcoin-core/gui-qml/commit/b8cf76c7c2279966c898a244d08de3757bd0eb4e)to[ 00f2a6a](https://github.com/bitcoin-core/gui-qml/commit/00f2a6aad5c04d00fae63f27cee9dea75e730ab2):
  - rebased with main

johnny9 commented:
I think I prefer this option as well after playing with both.  It is also easier to implement so that helps.

GBKS commented:
I just tested out the both options, the one here and the first one in #191. Is there any way to combine the two? Logic would be that below `minStickyHeight`, the logic here applies where the button is just part of the content and scrolls along with it. And above `minStickyHeight`, the button is sticky at the bottom of the screen?

johnny9 commented:
> I just tested out the both options, the one here and the first one in #191. Is there any way to combine the two? Logic would be that below `minStickyHeight`, the logic here applies where the button is just part of the content and scrolls along with it. And above `minStickyHeight`, the button is sticky at the bottom of the screen?
> 
> And thank you for the effort of setting up both of these PRs.



> I just tested out the both options, the one here and the first one in #191. Is there any way to combine the two? Logic would be that below `minStickyHeight`, the logic here applies where the button is just part of the content and scrolls along with it. And above `minStickyHeight`, the button is sticky at the bottom of the screen?
> 
> And thank you for the effort of setting up both of these PRs.

https://github.com/bitcoin-core/gui-qml/pull/209 here is an attempt at combining option 1 and 2.  I still need to work out some of the numbers to get the layout margins correct on both the "inside" and "outside" button but I set the minimum height to 300 pixels where it will switch from option1 to option2.  Testing on Desktop for quicker iteration.

jarolrod reviewed (commented):
This one works the best with tiling window managers:
![194](https://user-images.githubusercontent.com/23396902/208794552-88e70c35-a285-4173-af01-8d5ca5888a2f.png)

The other two options don't work well in split screen view, but this option works the best out of them, still not perfect though as shown by the screenshot below:

![Screenshot_20221220-175543](https://user-images.githubusercontent.com/23396902/208794739-53a7953c-72aa-45f5-895d-1560ac024c4c.jpg)

johnny9 commented:
Ok I'll focus on getting this version corrected.  Likely just have something off with the content height or the button anchor.  I think this option is the safest due to the simplicity of it. The layouts get difficult for the other options at the extremes

johnny9 commented:
Corrected the height calculations on mobile so the button hopefully doesn't overlap anymore.

shaavan reviewed (commented):
I experimented with decreasing the minimum window size to 200 on my mac to simulate the event when the content on the screen is longer than the height of the application.

And while using the scroll, I was able to observe some stuttering motions. I observed the stuttering at both 120 fps and 60 fps.

**Recording:**

https://user-images.githubusercontent.com/85434418/209706612-da05d8c3-3f77-4d12-923d-3d24d9aa65f9.mov

hebasto commented:
As https://github.com/bitcoin-core/gui-qml/pull/191 has been closed, I've updated this PR title.

jarolrod reviewed (approved):
ACK d17e30961ed3d0c8596ce0a8f00cb94750507e2c

I've tested on Android Tablet + Phone as well as on a tiling window manager. This works exactly as we'd want it to.

There's one issue with the contentHeight logic that doesn't have to be addressed as of now since it's really an edge case:

On a tiling window manager, if you shrink the window width so that the content needs to collapse and take up more vertical space, the contentHeight doesn't seem to be updated when the page's elements take up more vertical space due to the change in the width. Illustrated below, the continue button is barely clickable and not in view:

![scrollview-issue](https://user-images.githubusercontent.com/23396902/212595860-c725b903-f867-4d9b-812a-cf351ec84780.png)

This doesn't really apply to Android as of yet, because android doesn't seem to respect width changes whatsoever; but it would apply when we update the app logic to work with width changes on android.
